gpjax/parameters.py

```diff
@@ -1,27 +1,51 @@
 import typing as tp
 import warnings
+from collections import namedtuple
 from copy import deepcopy
 
 import distrax as dx
 import jax
 import jax.numpy as jnp
+from chex import dataclass
 from jaxtyping import f64
 from tensorflow_probability.substrates.jax import distributions as tfd
 
 from .config import get_defaults
+from .utils import merge_dictionaries
 
 Identity = dx.Lambda(lambda x: x)
 
 
 ################################
 # Base operations
 ################################
-def initialise(obj) -> tp.Tuple[tp.Dict, tp.Dict, tp.Dict]:
+@dataclass
+class ParameterState:
+    """The state of the model. This includes the parameter set and the functions that allow parameters to be constrained and unconstrained."""
+
+    params: tp.Dict
+    trainables: tp.Dict
+    constrainers: tp.Dict
+    unconstrainers: tp.Dict
+
+    def unpack(self):
+        return self.params, self.trainables, self.constrainers, self.unconstrainers
```

Comment:
I like the neatness of having all of these together. But,  similar to my `InferenceState` comment, what was the thinking behind a `dataclass` instead of something simpler and immutable such as a named tuple.

Reply:
Resolving as discussion can be held in the earlier comment.